nixos/modules/services/networking/netbird-server.nix

```diff
@@ -0,0 +1,423 @@
+      NETBIRD_AUTH0_DOMAIN = "";
+      NETBIRD_USE_AUTH0 = false;
+
+      NETBIRD_AUTH_DEVICE_AUTH_ENDPOINT = "";
+    };
+
+    services.netbird-server.managementConfig = with cfg.settings; {
+      Stuns = mkDefault [{
+        Proto = "udp";
+        URI = "stun:${TURN_DOMAIN}:${builtins.toString TURN_PORT}";
+        Username = STUN_USERNAME;
+        Password = STUN_PASSWORD;
+      }];
+      TURNConfig = {
+        Turns = [{
+          Proto = "udp";
+          URI = "turn:${TURN_DOMAIN}:${builtins.toString TURN_PORT}";
+          Username = TURN_USER;
+          Password = TURN_PASSWORD;
+        }];
+        CredentialsTTL = "12h";
+        Secret = TURN_SECRET;
+        TimeBasedCredentials = false;
+      };
+      Signal = {
+        Proto = NETBIRD_SIGNAL_PROTOCOL;
+        URI = "${NETBIRD_DOMAIN}:${builtins.toString NETBIRD_SIGNAL_PORT}";
+        Username = "";
+        Password = null;
+      };
+      Datadir = "${stateDir}/data";
+      HttpConfig = {
+        Address = "127.0.0.1:${builtins.toString cfg.ports.management}";
+        AuthIssuer = NETBIRD_AUTH_AUTHORITY;
+        AuthAudience = NETBIRD_AUTH_AUDIENCE;
+        AuthKeysLocation = NETBIRD_AUTH_JWT_CERTS;
+        AuthUserIDClaim = NETBIRD_AUTH_USER_ID_CLAIM;
+        OIDCConfigEndpoint = NETBIRD_AUTH_OIDC_CONFIGURATION_ENDPOINT;
+      };
+      IdpManagerConfig = {
+        ManagerType = NETBIRD_MGMT_IDP;
+        ClientConfig = {
+          Issuer = NETBIRD_AUTH_AUTHORITY;
+          TokenEndpoint = NETBIRD_AUTH_TOKEN_ENDPOINT;
+          ClientID = NETBIRD_IDP_MGMT_CLIENT_ID;
+          ClientSecret = NETBIRD_IDP_MGMT_CLIENT_SECRET;
+          GrantType = NETBIRD_IDP_MGMT_GRANT_TYPE;
+        };
+        ExtraConfig = cfg.idpManagerExtraConfig;
+      };
+      DeviceAuthorizationFlow = mkIf cfg.enableDeviceAuthorizationFlow {
+        Provider = NETBIRD_AUTH_DEVICE_AUTH_PROVIDER;
+        ProviderConfig = {
+          Audience = NETBIRD_AUTH_DEVICE_AUTH_AUDIENCE;
+          Domain = NETBIRD_AUTH_AUTHORITY;
+          ClientID = NETBIRD_AUTH_DEVICE_AUTH_CLIENT_ID;
+          TokenEndpoint = NETBIRD_AUTH_TOKEN_ENDPOINT;
+          DeviceAuthEndpoint = NETBIRD_AUTH_DEVICE_AUTH_ENDPOINT;
+          Scope = builtins.concatStringsSep " " NETBIRD_AUTH_DEVICE_AUTH_SCOPE;
+          UseIDToken = NETBIRD_AUTH_DEVICE_AUTH_USE_ID_TOKEN;
+        };
+      };
+      PKCEAuthorizationFlow = {
+        ProviderConfig = {
+          Audience = NETBIRD_AUTH_AUDIENCE;
+          ClientID = NETBIRD_AUTH_CLIENT_ID;
+          ClientSecret = NETBIRD_AUTH_CLIENT_SECRET;
+          AuthorizationEndpoint = NETBIRD_AUTH_PKCE_AUTHORIZATION_ENDPOINT;
+          TokenEndpoint = NETBIRD_AUTH_TOKEN_ENDPOINT;
+          Scope = builtins.concatStringsSep " " NETBIRD_AUTH_SUPPORTED_SCOPES;
+          RedirectURLs = NETBIRD_AUTH_PKCE_REDIRECT_URLS;
+          UseIDToken = NETBIRD_AUTH_PKCE_USE_ID_TOKEN;
+        };
+      };
+    };
+
+    services.nginx.virtualHosts = {
```

Comment:
Can you guard this behind a new enable option? Some users might prefer to use a different reverse proxy (or proxy via another machine) so adding this unconditionally is probably not desirable.